### PR TITLE
Refactor usage of EverestConfig in everest server functionality

### LINKED ISSUE #9182: Limit the use of EverestConfig in the Server related function signatures

### PR DESCRIPTION
**Pull request by DanSava**
**Issue**
Resolves #9182 

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/everest/detached/__init__.py [hunk 1]
```diff
@@ -54,35 +52,7 @@ async def start_server(config: EverestConfig, debug: bool = False) -> Driver:
...
-        log_level=logging.INFO,
-        log_to_azure=True,
-    )
-
-    configure_logger(
-        name=__name__,
-        file_path=os.path.join(log_dir, "simulations.log"),
-        log_level=logging.INFO,
-    )
-
```
Comment (frode-aarstad) [inline src/everest/detached/__init__.py:76]: Are these logs now in everserver? or is everest_server.log removed

File: src/everest/detached/__init__.py [hunk 2]
```diff
@@ -225,24 +193,16 @@ def server_is_running(url: str, cert: bool, auth: Tuple[str, str]):
     return True
 
 
-def get_optimization_status(config: EverestConfig):
```
Comment (frode-aarstad) [inline src/everest/detached/__init__.py:228]: Was this function not used?

Comment (DanSava) [reply src/everest/detached/__init__.py:228]: No it was not

File: src/everest/detached/__init__.py (hunk 1 above)
Comment (DanSava) [reply src/everest/detached/__init__.py:76]: Nothing was logging to any of these logging endpoints so these files were always empty so I just removed them.

**Review by frode-aarstad — APPROVED** [review]
🚀

**Pull request merged**